src/eventer/eventer_impl.c

```diff
@@ -103,16 +104,49 @@ eventer_impl_t __eventer = NULL;
 mtev_log_stream_t eventer_err = NULL;
 mtev_log_stream_t eventer_deb = NULL;
 
-static int __default_queue_threads = 5;
-static int __loop_concurrency = 0;
-static mtev_atomic32_t __loops_started = 0;
+static uint32_t __default_queue_threads = 5;
+static uint32_t __total_loop_count = 0;
+static uint32_t __default_loop_concurrency = 0;
 static eventer_jobq_t __default_jobq;
 
-int eventer_loop_concurrency() { return __loop_concurrency; }
+struct eventer_pool_t {
+  char *name;
+  uint32_t __global_tid_offset;
+  uint32_t __loop_concurrency;
+  uint32_t __loops_started;
+};
+
+static eventer_pool_t default_pool = { "default", 0 };
+static mtev_hash_table eventer_pools;
+
+void eventer_pool_create(const char *name, int concurrency) {
+  /* We cannot create pool once we've initialized */
+  mtevAssert(eventer_impl_tls_data == NULL);
+
+  eventer_pool_t *np = calloc(1, sizeof(*np));
+  np->name = strdup(name);
+  np->__loop_concurrency = concurrency;
+  if(!mtev_hash_store(&eventer_pools, np->name, strlen(np->name), np)) {
+    mtevFatal(mtev_stderr, "Two eventer pools with the same name.\n");
+  }
+}
+
+const char *eventer_pool_name(eventer_pool_t *pool) {
+  return pool->name;
+}
+
+eventer_pool_t *eventer_pool(const char *name) {
+  void *vptr;
+  if(mtev_hash_retrieve(&eventer_pools, name, strlen(name), &vptr))
+    return (eventer_pool_t *)vptr;
+  return NULL;
+}
+
+int eventer_loop_concurrency() { return __total_loop_count; }
```

Comment:
I think this should return `default_pool.__loop_concurrency`, and we should have a new routine, `eventer_loop_concurrency_pool` to return the concurrency for a specific pool?